parsely/src/main/java/com/parsely/parselyandroid/SendEvents.kt

```diff
@@ -0,0 +1,57 @@
+package com.parsely.parselyandroid
+
+import com.parsely.parselyandroid.JsonSerializer.toParselyEventsPayload
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+internal class SendEvents(
+    private val flushManager: FlushManager,
+    private val localStorageRepository: LocalStorageRepository,
+    private val parselyAPIConnection: ParselyAPIConnection,
+    private val scope: CoroutineScope
+) {
+
+    private val mutex = Mutex()
+
+    operator fun invoke(isDebug: Boolean) {
+        scope.launch {
+            mutex.withLock {
+                val eventsToSend = localStorageRepository.getStoredQueue()
+
+                if (eventsToSend.isEmpty()) {
+                    flushManager.stop()
+                    return@launch
+                }
+                ParselyTracker.PLog("Sending request with %d events", eventsToSend.size)
+
+                val jsonPayload = toParselyEventsPayload(eventsToSend)
+
+                ParselyTracker.PLog("POST Data %s", jsonPayload)
```

Comment:
Nitpick: These can probably go into the `else` clause - although if you do that you might want to add the number of events to the log in `isDebug` clause.

A further improvement might be to return from `isDebug` which should help with the readability. It'd also be more consistent with the prior `if (eventsToSend.isEmpty())` check.

Reply:
Sure, good point 👍 addressed in 269a9d6 and 409c42d

Reply:
Right! 🙂 Thanks, improves readability. Addressed in: 2c0ce6e